docs/pages/misc-pages/upgrade-guide.md

```diff
@@ -0,0 +1,152 @@
+the `exports` field of `@concordium/rust-bindings` to make it possible to select
+only parts of the SDK to include in your application.
+Furthermore, the SDK is now published as an ES module, making it possible to
+eliminate dead code.
+
+For **TypeScript** users, at least typescript version 5 is required along with
+the setting `compilerOptions.moduleResultion` to `"bundler"` to match the
+resolution strategy of modern bundlers.
+
+The following entrypoints are made available for consumers of
+`@concordium/web-sdk`:
+
+- `@concordium/web-sdk` exposes the full API of the SDK.
+This can safely be used by projects built with ES modules.
+- `@concordium/web-sdk/cis0` entrypoint exposes functionality for working with
+contracts adhering to the
+[CIS-0](https://proposals.concordium.software/CIS/cis-0.html) standard.
+- `@concordium/web-sdk/cis2` entrypoint exposes functionality for working with
+contracts adhering to the
+[CIS-2](https://proposals.concordium.software/CIS/cis-2.html) standard.
+- `@concordium/web-sdk/cis4` entrypoint exposes functionality for working with
+contracts adhering to the
+[CIS-4](https://proposals.concordium.software/CIS/cis-4.html) standard.
+- `@concordium/web-sdk/grpc` entrypoint exposes the grpc client for interacting
+with a nodes GRPCv2 interface.
+- `@concordium/web-sdk/id` entrypoint exposes functionality for working with
+ID proofs.
+- `@concordium/web-sdk/json-rpc` entrypoint exposes the **(deprecated)**
+json-rpc client for interacting with a nodes GPRCv1 interface.
+- `@concordium/web-sdk/schema` entrypoint exposes functionality for working
+with smart contract schemas, i.e.(de)serializing types using a smart
+contract schema.
+  - This uses the wasm entrypoint at `@concordium/rust-bindings/dapp`.
+- `@concordium/web-sdk/types` entrypoint exposes functionality for working
+with concordium domain types.
+- `@concordium/web-sdk/wasm` entrypoint exposes a variety of functionality for
+working with concordium domain types, which requires WASM.
+  - This uses the wasm entrypoint at `@concorodium/rust-bindings/wallet`.
+- `@concordium/web-sdk/web3-id` entrypoint exposes functionality for working
+with web3-id proofs.
+
+### NodeJS
+
+The `@concordium/web-sdk` module is not compatible with
+node versions <16 and is published only as an ES module.
+This, in turn, requires users to also run applications as ES modules.
+
+The easiest way to run your node application as an ES module, is by setting
+the `type` field of `package.json` to be set to `"module"`:
+
+```json
+{
+    ...
+    "type": "module",
+    ...
+}
+```
+
+Alternatively, files names with the extension `mjs` (or `mts` for TypeScript)
+are always handled as ES modules.
+
+For **TypeScript** users, at least typescript version 4.7 is required along
+with the setting `compilerOptions.moduleResultion` to `"node16"` or
+`"nodenext"` to match the resolution strategy of node version 16 and later.
+
+## Common SDK version 5 to 6 (Web 2->3) (Node 5->6)
```

Comment:
Maybe we should add here as well that these packages are merged into web-sdk 7 and forward

Reply:
Yes, good idea. I added a "notice" under the nodeJS header.